driver-core/src/main/com/mongodb/internal/connection/DefaultServerMonitor.java

```diff
@@ -112,7 +112,7 @@ public void start() {
     public void connect() {
         lock.lock();
         try {
-            condition.signal();
+            condition.signalAll();
```

Comment:
From what I can see, the RTT thread doesn't await this condition.  It calls `DefaultServerMonitor#waitForNext`, which just sleeps.  It's confusing because there are two waitForNext methods in the same file (also `DefaultServerMonitor.ServerMonitorRunnable#waitForNext`).

Reply:
Indeed, good catch!